File: kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        return if (input.contains(Regex("[A-Z]")) && input == input.toUpperCase()) {
           "Whoa, chill out!"
        } else if (input.endsWith("?")) {
            "Sure."
        } else if (input.isBlank()) {
            "Fine. Be that way!"
        } else {
            "Whatever."
        }
    }
}

File: scala/bob/src/main/scala/Bob.scala
object Bob {
  def response(statement: String): String = {
    if (isShouting(statement)) "Whoa, chill out!"
    else if (statement.endsWith("?")) "Sure."
    else if (statement.replaceAll("\\s*", "") .isEmpty) "Fine. Be that way!"
    else "Whatever."
  }

  private def isShouting(statement: String): Boolean = {
    statement == statement.toUpperCase() & statement.replaceAll("[a-zA-Z]", "") != statement
  }
}
